test/compflags/lydia/library/exportArray/exportFuncWith1Arr1Not.h

```diff
@@ -0,0 +1,6 @@
+#include <stdint.h>
+
+#include "chpl-external-array.h"
+
+void foo(chpl_external_array* x, int64_t otherArg);
```

Comment:
I'm surprised by the `*` here. I would have expected the struct to be passed directly.

Reply:
I believe this is due to the compiler seeing chpl_external_array is a record (albeit an extern one) and transforming the argument into a ref at resolution.  I can try to make it not do that for exported functions, but I assumed we wanted to keep that behavior.

Reply:
I don't think the compiler should be doing this for extern types and certainly not for extern types in exported functions. In resolveIntents, concreteIntentForArg has a check for fn having FLAG_EXTERN. What if you add FLAG_EXPORT to that check?

Reply:
Looks like that gets honored all the way until codegen, at which case it gets thrown out.  I'll look into why codegen is doing that

Reply:
at a guess, have a look at ArgSymbol::requiresCPtr

Reply:
I tried updating the return check at the end of `shouldChangeArgumentTypeToRef` (where that method is called) similarly to the resolveIntents change.  This resulted in a lot of problems at makeBinary due to I believe some calls defined in the runtime (I see chpl_main_argument and chpl_localeID_t involved, mostly the latter).

The chpl_localeID_t errors go away if I instead update argMustUseCPtr to check if the record is defined as extern, but the chpl_main_argument errors still remain - I'm not as worried when it is just that which gets affected, though, so I'm going to see if it compiles and runs when I update the uses of chpl_main_argument.

Lemme know your thoughts on this

Reply:
I think we should be passing around chpl_localeID_t by value where possible so I think we should be passing them by value.

Reply:
Modifying the use of chpl_main_argument, I was able to get my test compiling and running without a segfault.

Reply:
> I think we should be passing around chpl_localeID_t by value where possible too.

So do you think the modification for `shouldChangeArgumentTypeToRef` (keying off of if the function is exported, rather than if the type is an extern record) is the better call?

Reply:
Let's leave these changes out of this PR and put them in a new PR. They probably will need discussion from others too, so best to try to separate them.

Reply:
I opened issue #9898 to track this.  I'll send it to the larger group once you've had a chance to look at it